### PR TITLE
test(cronjobs): adding unit tests for timezones

### LINKED ISSUE #47202: CronJobs should support timezones
I couldn't find any discussion of this after searching for "cron timezone", "cronjob timezone", or "scheduledjob timezone".

The CronJob spec makes reference to https://en.wikipedia.org/wiki/Cron. That page suggests that cron would respect the timezone for a given user. The controller manager runs in a single time zone under a single user so I can't use different time zones for each job. I have jobs that run based on the schedule of external entities that observe daylight savings time. So, if I define that CronJob in UTC I will be forced to update that job from time to time (generally not something one remembers to do after just losing an hour of sleep).

I see two options for how this support might work in kubernetes:

1. Add some new field to the `CronJobSpec`, like `timezone: "Americas/Chicago"`.
2. Use an extended cron syntax that includes timezone, e.g. `30 6 * * 1 Europe/Stockholm`

### PR DESCRIPTION
**Pull request by cndoit18**
Signed-off-by: cndoit18 <cndoit18@outlook.com>

#### What type of PR is this?

/kind cleanup
/sig testing

#### What this PR does / why we need it:

I observed that the `github.com/robfig/cron` was upgraded a few months ago and it supports the ability to customize the time zone, so I added this part of the unit test

fixes: #47202 
relate: #102735
see: https://pkg.go.dev/github.com/robfig/cron/v3#hdr-Time_zones

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

**Comment by cndoit18** [comment]
I think the description of the official website should also be updated
image

**Comment by cndoit18** [comment]
/assign soltysh
Hi, @soltysh this is a minor change so that we can notice that it supports timezone handling now. 
Can you review it?
thank you very much.

File: pkg/apis/batch/validation/validation_test.go [hunk 1]
```diff
@@ -606,6 +606,22 @@ func TestValidateCronJob(t *testing.T) {
...
 			},
 		},
+		"basic scheduled job with timezone": {
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "mycronjob",
+				Namespace: metav1.NamespaceDefault,
+				UID:       types.UID("1a2b3c"),
+			},
+			Spec: batch.CronJobSpec{
+				Schedule:          "CRON_TZ=UTC * * * * ?",
```
Comment (cndoit18) [inline pkg/apis/batch/validation/validation_test.go:616]: see: https://pkg.go.dev/github.com/robfig/cron/v3#hdr-Time_zones

**Comment by cndoit18** [comment]
/assign soltysh liggitt
Hi, This is a minor change so that we can notice that it supports timezone handling now. 
Can you review it?
Thank you very much.

cc @liggitt @soltysh

**Comment by liggitt** [comment]
/unassign

feel free to reassign once this has been reviewed

**Comment by cndoit18** [comment]
Any update?

**Comment by soltysh** [comment]
/hold
/assign 
this is being discussed during next sig-apps on Nov 15th

**Comment by soltysh** [comment]
This will be marked as unsupported feature in https://github.com/kubernetes/kubernetes/pull/106455. I'll open separate PRs updating docs. 
/close

**Pull request closed without merging**